apps/src/studio/blocks.js

```diff
@@ -824,9 +824,16 @@ exports.install = function(blockly, blockInstallOptions) {
   ];
 
   generator.studio_setItemSpeed = function() {
+    var classParam = this.getFieldValue('CLASS');
+    if (classParam === RANDOM_VALUE) {
+      var allValues = skin.itemChoices.slice(0, -1).map(function(item) {
+        return item[1];
+      });
+      classParam = 'Studio.random([' + allValues + '])';
```

Comment:
Nice investigation/revision based on Brendan's comment!
Would it help readability to directly assign `classParam` to a random value from the `allValues` array here?

Reply:
Actually let me amend that. Brendan's explanation is more accurate. We are only setting this generator function once when install the blocks. If we randomly select an item here, that's the one that will be included in the generated code for each instance of the block where "random" is selected.

Reply:
Thanks to you both for the explanation!